server/lib/storage/database.js

```diff
@@ -54,13 +66,24 @@ export default class Database {
   }
 
   createPermission(permission) {
-    return this.provider
-      .create('permissions', permission);
+    return this.getPermissions()
+      .then(permissions =>
+        checkUnique(
+          permissions.filter(item => item.name === permission.name && item.applicationId === permission.applicationId),
+          `Permission with name "${permission.name}" is already exists.`
```

Comment:
this is not an error with a message field, that's why it was not showing the correct error message ;)